File: src/main/java/org/kane/domain/service/recipe_recource/coefficient/CoefficientService.java
package org.kane.domain.service.recipe_recource.coefficient;

import org.kane.database.entity.recipe_recource.Category;
import org.kane.domain.DTO.entityDTO.recipe_recource.coefficient.CoefficientCreateDTO;
import org.kane.domain.DTO.entityDTO.recipe_recource.coefficient.CoefficientEditDTO;
import org.kane.domain.DTO.entityDTO.recipe_recource.coefficient.CoefficientShowDTO;

public interface CoefficientService{
    CoefficientShowDTO addCoefficient(CoefficientCreateDTO coefficientShowDTO, Category category);

    CoefficientShowDTO editCoefficient(CoefficientEditDTO coefficientEditDTO);
}

File: src/app/service/recipe_resource/coefficient.service.ts
import {inject, Injectable} from '@angular/core';
import {HttpClient} from '@angular/common/http';
import {Endpoint} from '../../util/endpoint';
import CoefficientEditDTO from '../../DTO/entity_dto/recipe-recource/coefficient/coefficient-edit.dto';
import {Observable} from 'rxjs';
import CoefficientShowDTO from '../../DTO/entity_dto/recipe-recource/coefficient/coefficient-show.dto';

@Injectable({
  providedIn: 'root',
})
export class CoefficientService {
  private readonly http = inject(HttpClient);
  private readonly categoryAPI = new Endpoint('coefficient');

  editCoefficient(coefficient:CoefficientEditDTO):Observable<CoefficientShowDTO>{
    return this.http.patch<CoefficientShowDTO>(
      this.categoryAPI.builder()
        .points("edit")
        .build(),
      coefficient)
  }

}
